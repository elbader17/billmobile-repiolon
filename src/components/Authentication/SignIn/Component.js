import React from 'react';
import { Alert, StyleSheet, View } from 'react-native';
import { Input, Button } from "react-native-elements";


class SignIn extends React.Component {

    constructor(props) {
      super(props);
      this.state = {
        email: '',
        password: '@Am1234!',
        name:'',
        confirmPassword: '@Am1234!',
      };
    }

    handleSignIn = () => {
      const { email, password } = this.state;
      const { signIn } = this.props;
      signIn(email, password);
            
    }

    setEmail = (value) => this.setState({ email: value })
    setPassword = (value) => this.setState({ password: value })

    render() {
      return(
          <View>
            <Input
              value={this.state.email}
              label="Email"
              leftIcon={{ type: 'font-awesome', name: 'envelope' }}
              onChangeText={this.setEmail}
              placeholder="my@email.com"
            />
            <Input
              value={this.state.password}
              label="Password"
              leftIcon={{ type: 'font-awesome', name: 'lock' }}
              onChangeText={this.setPassword}
              placeholder="p@ssw0rd123"
              secureTextEntry
            />
            <Button
              testID={'submitSignIn'}
              id='submitSignIn'
              title='Submit'
              onPress={ this.handleSignIn }
              
            />
        </View>
      )
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
});


export default SignIn;
