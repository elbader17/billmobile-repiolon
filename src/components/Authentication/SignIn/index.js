import React from 'react';
import { StyleSheet, View } from 'react-native';
import { Input, Button } from "react-native-elements";
import { signIn } from '../../../app/authentication/actions';

class SignIn extends React.Component {

    constructor(props) {
      super(props);
      this.state = {
        email: '',
        password: '@Am1234!',
        name:'',
        confirmPassword: '@Am1234!',
      };
    }

    handleSignIn = () => {
      const { email, password } = this.state;
      const { dispatch } = this.props;
      dispatch(signIn(email, password));
    }

    setEmail = (value) => this.setState({ email: value })
    setPassword = (value) => this.setState({ password: value })

    render() {
      return(
          <View>
            <Input
              label="Email"
              leftIcon={{ type: 'font-awesome', name: 'envelope' }}
              onChangeText={this.setEmail}
              placeholder="my@email.com"
            />
            <Input
              label="Password"
              leftIcon={{ type: 'font-awesome', name: 'lock' }}
              onChangeText={this.setPassword}
              placeholder="p@ssw0rd123"
              secureTextEntry
            />
            <Button
              title='Submit'
              onPress={ this.handleSignIn }
            />
        </View>
      )
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
});


export default SignIn;
