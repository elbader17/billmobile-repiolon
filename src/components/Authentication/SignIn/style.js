import { StyleSheet } from 'react-native';
import { COLORS } from '../../../constants/colors';
import { FONTS } from '../../../constants/fonts';
import { METRICS } from '../../../constants/metrics';

const styles = StyleSheet.create({
  container:{
    width: METRICS.screenWidth,
    height: METRICS.screenHeight
  },
  container2:{
    alignItems: 'center',
    paddingHorizontal: 45,
    paddingTop: 15
  },
  textBoxBtnHolder: {
    position: 'relative',
    alignSelf: 'stretch',
    justifyContent: 'center',
    marginTop: 5
  },
  textBox: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size18,
    alignSelf: 'stretch',
    height: 45,
    paddingRight: 45,
    paddingLeft: 8,
    borderWidth: 1,
    paddingVertical: 0,
    borderColor: COLORS.gray,
    borderRadius: 2
  },
  visibilityBtn: {
    position: 'absolute',
    right: 3,
    height: 40,
    width: 35,
    padding: 5
  },
  btnImage: {
    resizeMode: 'contain',
    height: '100%',
    width: '100%',
    opacity: 0.4
  },
  submit: {
    backgroundColor: COLORS.red,
    width: 266,
    height: 40,
    borderRadius: 2
  },
  submitDisabled: {
    backgroundColor: COLORS.redLight,
    width: 266,
    height: 40,
    borderRadius: 2
  },
  submitText: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size14,
    color: COLORS.white
  },
  textRegister: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size11,
    color: COLORS.grayDark,
    paddingVertical: 15
  },
  red: {
    color: COLORS.red
  },
  textFooterA: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size11,
    color: COLORS.grayDark,
    marginTop: 110
  },
  textFooterB: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size11,
    color: COLORS.grayDark,
    fontWeight: 'bold'
  }
});

export default styles;