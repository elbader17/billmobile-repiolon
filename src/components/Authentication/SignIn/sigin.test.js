import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SignIn from './Component';

describe('Rendering Component SigIn', () => {
  test('renders correctly SignIn', () => {
    const component = renderer.create(<SignIn />).toJSON();
    expect(component).toMatchSnapshot();
  })
});

describe('Rendering Component and call  handeleSigIn', () => {
  it('call the handleSignIn', () => {
    const clickFn = jest.fn();
    const wrapper = shallow(<SignIn  signIn={clickFn} />);
    const button = wrapper.findWhere(node => node.prop('testID') === 'submitSignIn');
    button.simulate('press');
    expect(clickFn).toBeCalled();
  })
});

describe('Test to validate data of form ',() => {
  it('Test Password with atributes disabled true, initial state', () => {
    const clickFn = jest.fn();
    const wrapper = shallow(<SignIn signIn={clickFn} />);
    const button = wrapper.findWhere(node => node.prop('testID') === 'submitSignIn');
    expect(button.props().disabled).toEqual(true);
  })
  it('Test Password with atributes disabled true', () => {
   const wrapper = shallow(<SignIn />);
    wrapper.setState({ password: 'agustin' });
    wrapper.setState({ email: 'agustin.com' });
    const button = wrapper.findWhere(node => node.prop('testID') === 'submitSignIn');
    expect(button.props().disabled).toEqual(true);
  })
  it('Test Password with atributes disabled false', () => {
    const wrapper = shallow(<SignIn />);
    wrapper.setState({ email: 'agustin@gmail.com' });
    wrapper.setState({ password: '@Am1234-' });
    const button = wrapper.findWhere(node => node.prop('testID') === 'submitSignIn');
    expect(button.props().disabled).toEqual(false);
   })
});