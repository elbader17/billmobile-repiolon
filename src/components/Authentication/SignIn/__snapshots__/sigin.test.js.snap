// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Component SigIn renders correctly SignIn 1`] = `
<View>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#86939e",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Email
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
            underlayColor="white"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 24,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </View>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="my@email.com"
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#86939e",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Password
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
            underlayColor="white"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 24,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </View>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="p@ssw0rd123"
        secureTextEntry={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value="@Am1234!"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "borderRadius": 3,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="submitSignIn"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2089dc",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          Submit
        </Text>
      </View>
    </View>
  </View>
</View>
`;
